lib/puppet/resource/capability_finder.rb

```diff
@@ -26,7 +26,12 @@ def self.find(environment, code_id, cap)
       raise Puppet::DevError, 'PuppetDB is not available'
     end
 
-    resources = search(environment, nil, cap)
+    resources = search(nil, nil, cap)
+
+    if resources.size > 1
+      Puppet.debug "Found multiple resources when looking up capability #{cap}, filtering by environment #{environment}"
+      resources = search(environment, nil, cap)
```

Comment:
Because the environment is indicated by a tag, we should be able to filter the resources we just retrieved, rather than making another query. We make a second query for the code id case just because the code id isn't returned as part of the resource.